### PR TITLE
Add remaining evolutions

### PR DESCRIPTION
**Pull request by 3geek14**
This is preparing for #357

### CODE REVIEW

File: common/src/utils/pokemon-utils/evolution-utils.ts
Comment (SleepAPI) [inline common/src/utils/pokemon-utils/evolution-utils.ts:1]: Really cool idea! Missing tests for this

**Review by SleepAPI — CHANGES_REQUESTED** [review]
Just missing testing on the new evolutions-utils. Really great stuff. here

File: common/src/utils/pokemon-utils/evolution-utils.ts
Comment (3geek14) [reply common/src/utils/pokemon-utils/evolution-utils.ts:1]: I have integration tests in pokemon.test.ts, and I think those are better tests than what we can get from unit testing these functions. Even as these functions grow (from #359, and possibly if we add other stuff later), I think they're simple enough that they might not benefit from unit tests.

I've added two tests. I'm happy to remove them if you don't like them. If there's another type of unit test you'd rather see, let me know?

Comment (3geek14) [reply common/src/utils/pokemon-utils/evolution-utils.ts:1]: Note that the [code coverage report](https://app.codecov.io/gh/SleepAPI/SleepAPI/pull/358?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=McMomo) from yesterday shows that the only untested line is in index.ts.

Comment (SleepAPI) [reply common/src/utils/pokemon-utils/evolution-utils.ts:1]: Yes it was tested indirectly, but I personally prefer testing directly too for several reasons, but some bigger reasons are traceability and code standard. By standard we always accompany new functionality by test files. When it comes to traceability if the test in pokemon.test.ts fails you'd have to debug why, in practice it's probably quite easy and this is a smaller issue here, but if we add tests in evolution-utils.test.ts and that specifically fails then it's even easier to pinpoint the exact failing function.

I saw you added tests though and they look good to me!

**Review by SleepAPI — APPROVED** [review]

**Pull request merged**